contracts/market/FifoQueue.sol

```diff
@@ -0,0 +1,57 @@
+
+pragma solidity ^0.4.24 ;
+import "../../node_modules/zeppelin-solidity/contracts//math/SafeMath.sol";
+
+contract FifoQueue {
+  using SafeMath for uint256; //todo jaycen PRELAUNCH - make sure we use this EVERYWHERE its needed
+
+  mapping(uint256 => uint256) fifoQueue;
+  uint256 first = 1;
+  uint256 last = 0;
+
+  function push(uint256 tokenId) public {
+    last = last.add(1);
+    fifoQueue[last] = tokenId;
+  }
+
+  function pop() public returns (uint256 tokenId) {
```

Comment:
`pop()` works by iterating through the length of the FifoQueue. It peeks at the first element, checking to see if it has been removed. If it has, we move the front of the queue. If it hasn't, we delete that item from the queue, move the front of the queue, and then return the tokenId.